src/vs/workbench/contrib/terminal/browser/media/shellIntegration-rc.zsh

```diff
@@ -14,17 +14,19 @@ fi
 # as disable it by unsetting the variable.
 VSCODE_SHELL_INTEGRATION=1
 
+# Set HISTFILE to the default location before the user's .zshrc is sourced
+# to prevent the value of HISTFILE from defaulting to VSCODE_ZDOTDIR/.zsh_history
+if [[  "$VSCODE_INJECTION" == "1"  && -f $USER_ZDOTDIR/.zsh_history && -f $HISTFILE ]]; then
```

Comment:
I think we want to remove the middle condition and do it regardless (eg. first time zsh launches it may not exist?)

Reply:
yep 👍🏼 good call

Reply:
I think we should also remove the second check IE `-f $HISTFILE`

Reply:
yep removed that and all of the cases still work